rpc/utils.go

```diff
@@ -45,6 +45,21 @@ func (s *rpcServer) broadcastMessage(msgBz []byte) error {
 	return nil
 }
 
+// getQueryHeight returns either the height supplied or if it is equal to 0
+// the most recent block height that has been commited. As the current consensus height
+// is one above this, and if used in certain queries will return an error as the height
+// has not been commited yet
+func (s *rpcServer) getQueryHeight(height int64) int64 {
+	if height == 0 {
+		currentHeight := int64(s.GetBus().GetConsensusModule().CurrentHeight())
+		if currentHeight > 0 {
+			currentHeight -= 1
+		}
+		return currentHeight
+	}
+	return height
```

Comment:
Just a NIT that makes code easier to read. It's a general trick if you ever find yourself doing `if {very large block } `
```suggestion
	if height != 0 {
		return height
	}
	currentHeight := int64(s.GetBus().GetConsensusModule().CurrentHeight())
	if currentHeight > 0 {
		currentHeight -= 1
	}
	return currentHeight
```